embulk-jruby-extension/src/main/java/org/jruby/util/ParsedValues.java

```diff
@@ -0,0 +1,406 @@
+            offset = ZONE_OFFSET_MAP.get(z);
+            if (dst) {
+                offset += 3600;
+            }
+            return offset;
+        }
+
+        if (z.startsWith("gmt") || z.startsWith("utc")) {
+            z = z.substring(3, z.length()); // remove "gmt" or "utc"
+        }
+
+        if (!isSign(z.charAt(0))) {
+            // if z doesn't have "+" or "-", invalid
+            return offset;
+        }
+
+        boolean sign = false;
+        sign = z.charAt(0) == '+';
+        z = z.substring(1, z.length());
+
+        int hour = 0, min = 0, sec = 0;
+        if (z.contains(":")) {
+            String[] splited = z.split(":");
+            if (splited.length == 2) {
+                hour = parseInt(splited[0]);
+                min = parseInt(splited[1]);
+            } else {
+                hour = parseInt(splited[0]);
+                min = parseInt(splited[1]);
+                sec = parseInt(splited[2]);
+            }
+        } else if (z.contains(",") || z.contains(".")) {
+            // TODO min = Rational(fr.to_i, 10**fr.size) * 60
+            String[] splited = z.split("[\\.,]");
+            hour = parseInt(splited[0]);
+            min = (int)(parseInt(splited[1]) * 60 / Math.pow(10, splited[1].length()));
+
+        } else {
+            int len = z.length();
+            if (len % 2 != 0) {
+                if (len >= 1)
+                    hour = parseInt(z.substring(0, 1));
+                if (len >= 3)
+                    min = parseInt(z.substring(1, 3));
+                if (len >= 5)
+                    sec = parseInt(z.substring(3, 5));
+
+            } else {
+                if (len >= 2)
+                    hour = parseInt(z.substring(0, 2));
+                if (len >= 4)
+                    min = parseInt(z.substring(2, 4));
+                if (len >= 6)
+                    sec = parseInt(z.substring(4, 6));
+            }
+        }
+
+        offset = hour * 3600 + min * 60 + sec;
+        return sign ? offset : -offset;
+    }
+
+    private static int parseInt(String text)
+    {
+        int v = 0;
+        for (int i = 0; i < text.length(); i++) {
+            char c = text.charAt(i);
+            if (!isDigit(c)) {
+                break;
+            } else {
+                v = v * 10 + toInt(c);
+            }
+        }
+        return v;
+    }
+
+    int mday = Integer.MIN_VALUE;
+    int wday = Integer.MIN_VALUE;
+    int cwday = Integer.MIN_VALUE;
+    int yday = Integer.MIN_VALUE;
+    int cweek = Integer.MIN_VALUE;
+    int cwyear = Integer.MIN_VALUE;
+    int min = Integer.MIN_VALUE;
+    int mon = Integer.MIN_VALUE;
+    int hour = Integer.MIN_VALUE;
+    int year = Integer.MIN_VALUE;
+    int sec = Integer.MIN_VALUE;
+    int wnum0 = Integer.MIN_VALUE;
+    int wnum1 = Integer.MIN_VALUE;
+
+    String zone = null;
+
+    int sec_fraction = Integer.MIN_VALUE; // Rational
+    int sec_fraction_size = Integer.MIN_VALUE;
```

Comment:
Do you need sec_fraction + sec_fraction_size? Why don't you have nanoseconds only?
`Rational(1, 10) == Rational(10, 100)` is true.